src/call_state_machine.rs

```diff
@@ -0,0 +1,131 @@
+#[derive(Debug)]
+pub enum CallState {
+    Idle,
+    OutgoingCall,
+    IncomingCall,
+    InProgressCall,
+}
+
+#[derive(Debug)]
+pub enum CallSwitchEdge {
+    NoEdge,
+    Active,
+    Inactive,
+}
+
+pub enum CallSwitchState {
+    Active,
+    Inactive,
+}
+
+pub struct CallSwitch {
+    current_state: Option<CallSwitchState>,
+}
+
+impl CallSwitch {
+    fn new() -> Self {
+        CallSwitch {
+            //TODO read GPIO pin? Or set to None?
+            current_state: None,
+        }
+    }
+
+    fn dispatch(mut self, new_switch_state: CallSwitchState) {
+        use CallSwitchState::*;
+
+        let edge: CallSwitchEdge = match (self.current_state, new_switch_state) {
+            /* First call to dispatch */
+            (None, _) => CallSwitchEdge::NoEdge,
```

Comment:
why not have the first call immediately be an edge?

Reply:
cause then you turn it on and it starts calling? lol

Reply:
i would expect it to start calling immediately if the switch is on